src/Microsoft.Diagnostics.Monitoring.WebApi/Stacks/NameFormatter.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Text;
+
+#if STARTUPHOOK
+namespace Microsoft.Diagnostics.Monitoring.StartupHook
+#else
+#nullable enable
+
+namespace Microsoft.Diagnostics.Monitoring.WebApi.Stacks
+#endif
+{
+    internal sealed class NameFormatter
+    {
+        public const string UnknownModule = "UnknownModule";
+        public const string UnknownClass = "UnknownClass";
+
+        private const string ArrayType = "_ArrayType_";
+        private const string CompositeType = "_CompositeType_";
+
+        private const char NestedSeparator = '+';
+        private const char GenericStart = '[';
+        private const char GenericSeparator = ',';
+        private const char GenericEnd = ']';
+
+        public static void BuildClassName(StringBuilder builder, NameCache cache, FunctionData functionData)
+        {
+            if (functionData.ParentClass != 0)
+            {
+                BuildClassName(builder, cache, functionData.ParentClass);
+            }
+            else
+            {
+                BuildClassName(builder, cache, functionData.ModuleId, functionData.ParentToken);
+            }
+        }
+
+        public static void BuildClassName(StringBuilder builder, NameCache cache, ulong classId)
+        {
+            string className = UnknownClass;
+            if (cache.ClassData.TryGetValue(classId, out ClassData? classData))
```

Comment:
I noticed that in the existing code we aren't using nullable when we try to get a value (e.g. `ClassData` instead of `ClassData?`), but we are in all of the new code it looks like - what's the reason?

Reply:
The StartupHook project has nullability enforcement enabled whereas the existing projects do not. I don't want to go through the rest of dotnet-monitor and make that kind of change as part of this work.